test/integration/translation_failures_test.go

```diff
@@ -0,0 +1,130 @@
+//go:build integration_tests
+// +build integration_tests
+
+package integration
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/annotations"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane"
+	kongv1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1"
+	"github.com/kong/kubernetes-ingress-controller/v2/pkg/clientset"
+)
+
+// TestTranslationFailures ensures that proper warning Kubernetes events are recorded in case of translation failures
+// encountered.
+func TestTranslationFailures(t *testing.T) {
+	testCases := []struct {
+		name string
+		// translationFailureTrigger should create objects that trigger translation failure and return the objects
+		// that we expect translation failure warning events to be created for.
+		translationFailureTrigger func(ns string) []client.Object
+	}{
+		{
+			name: "invalid CA secret",
+			translationFailureTrigger: func(ns string) []client.Object {
+				createdSecret, err := env.Cluster().Client().CoreV1().Secrets(ns).Create(ctx, invalidCASecret(ns), metav1.CreateOptions{})
+				require.NoError(t, err)
+
+				return []client.Object{createdSecret}
+			},
+		},
+		{
+			name: "invalid CA secret referred by a plugin",
+			translationFailureTrigger: func(ns string) []client.Object {
+				createdSecret, err := env.Cluster().Client().CoreV1().Secrets(ns).Create(ctx, invalidCASecret(ns), metav1.CreateOptions{})
+				require.NoError(t, err)
```

Comment:
I suggest to capture the subtest's `t *testing.T` so that this `require` won't abort the whole test in case of a failure.

Reply:
Yup, good catch 🙇: [c4a6122](https://github.com/Kong/kubernetes-ingress-controller/pull/3125/commits/c4a61220355922b891663591830aaca3e9dbcef1)